specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingProperty.json

```diff
@@ -65,182 +95,553 @@
+            "$ref": "./examples/billingPropertyPatchSubscriptionServiceUsageAddress.json"
           }
         }
       }
     }
   },
   "definitions": {
     "BillingProperty": {
+      "description": "A billing property.",
+      "type": "object",
       "allOf": [
         {
           "$ref": "./types.json#/definitions/ArmResource"
         }
       ],
-      "description": "A billing property.",
       "properties": {
         "properties": {
-          "$ref": "#/definitions/BillingPropertyProperties",
-          "description": "A billing property.",
+          "description": "The properties of a(n) BillingProperty",
+          "allOf": [
+            {
+              "$ref": "#/definitions/BillingPropertyProperties"
+            }
+          ],
           "x-ms-client-flatten": true
         }
       },
-      "type": "object"
+      "x-ms-azure-resource": true
     },
     "BillingPropertyProperties": {
-      "description": "The billing property.",
+      "description": "A billing property.",
+      "type": "object",
       "properties": {
-        "accountAdminNotificationEmailAddress": {
-          "description": "The email address on which the account admin gets all Azure notifications.",
+        "billingAccountAgreementType": {
+          "description": "The type of agreement.",
+          "enum": [
+            "Unknown",
+            "MicrosoftCustomerAgreement",
+            "EnterpriseAgreement",
+            "MicrosoftOnlineServicesProgram",
+            "MicrosoftPartnerAgreement"
+          ],
+          "type": "string",
           "readOnly": true,
-          "type": "string"
+          "x-ms-enum": {
+            "name": "AgreementType",
+            "modelAsString": true
+          }
         },
         "billingAccountDisplayName": {
-          "description": "The name of the billing account to which the subscription is billed.",
-          "readOnly": true,
-          "type": "string"
+          "description": "The name of the billing account.",
+          "type": "string",
+          "readOnly": true
         },
         "billingAccountId": {
-          "description": "The ID of the billing account to which the subscription is billed.",
+          "description": "The fully qualified ID that uniquely identifies a billing account.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingAccountNotificationEmailAddress": {
+          "description": "Notification email address for legacy account. Available for agreement type Microsoft Online Services Program.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingAccountSoldToCountry": {
+          "description": "The address of the individual or organization that is responsible for the billing account.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingAccountStatus": {
+          "description": "The current status of the billing account.",
```

Comment:
What does Extended mean in this case? Is that EA specific? If so, can this be moved into an EA-specific enum under enrollmentDetails?

Reply:
@ramkd12 @amberbhargava to comment here

Reply:
Moving this tatus to an EA-specific enum complicates the API for the customer. If we are supporting EA accounts, then we should show their status here.